test/pylib: servers_add: support list of property_files

bhalevy opened the pull request:
So that a multi-dc/multi-rack cluster can be populated
in a single call.

* Enhancement, no backport required

mykaul commented:
CCM changes the snitch, if there's a topology that SimpleSnitch doesn't support. Shouldn't we do the same and move to the file based snitch?

bhalevy commented:
> CCM changes the snitch, if there's a topology that SimpleSnitch doesn't support. Shouldn't we do the same and move to the file based snitch?

This PR is for scylla unit tests so CCM isn't involved.
In pylib we use the file-based snitch if `property_file` is provided by the caller, and `SimpleSnitch` if it isn't.
This actually helps us find cases where the property file was used on some of the nodes while it wasn't provided on others.
If we hide that by automating the property file in the lower layers I'm afraid it could end up with inconsistent topologies.

dawmd reviewed (approved):
LGTM. Thanks for covering this, Benny!

mykaul commented:
> > CCM changes the snitch, if there's a topology that SimpleSnitch doesn't support. Shouldn't we do the same and move to the file based snitch?
> 
> This PR is for scylla unit tests so CCM isn't involved. In pylib we use the file-based snitch if `property_file` is provided by the caller, and `SimpleSnitch` if it isn't. This actually helps us find cases where the property file was used on some of the nodes while it wasn't provided on others. If we hide that by automating the property file in the lower layers I'm afraid it could end up with inconsistent topologies.

Sound like if property file is NOT used, but a topology IS used, we'll end up in a strange configuration - that may not be supported. CCM fixed it, which is why I mentioned it.

scylladb-promoter commented:
#### :red_circle: CI State: FAILURE

:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/16117/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/16117/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: cluster/auth_cluster/test_auth_no_quorum
:small_blue_diamond: cluster/auth_cluster/test_auth_raft_command_split
:small_blue_diamond: cluster/auth_cluster/test_auth_v2_migration
:small_blue_diamond: cluster/auth_cluster/test_raft_service_levels
:small_blue_diamond: cluster/test_aggregation
:small_blue_diamond: cluster/test_automatic_cleanup
:small_blue_diamond: cluster/test_cluster_features
:small_blue_diamond: cluster/test_concurrent_schema
:small_blue_diamond: cluster/test_gossiper
:small_blue_diamond: cluster/test_mutation_schema_change
:small_blue_diamond: cluster/test_mv
:small_blue_diamond: cluster/test_random_tables
:small_blue_diamond: cluster/test_replace_alive_node
:small_blue_diamond: cluster/test_topology_rejoin
:small_blue_diamond: cluster/test_topology_schema
:x: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/16117/)
#### Failed Tests (7/37895):
- [test_joining_old_node_fails.debug.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/16117/testReport/junit/cluster/test_raft_cluster_features/Tests___Unit_Tests___test_joining_old_node_fails_debug_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_joining_old_node_fails.debug.1)
- [test_joining_old_node_fails.debug.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/16117/testReport/junit/cluster/test_raft_cluster_features/Tests___Unit_Tests___test_joining_old_node_fails_debug_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_joining_old_node_fails.debug.2)
- [test_joining_old_node_fails.debug.3](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/16117/testReport/junit/cluster/test_raft_cluster_features/Tests___Unit_Tests___test_joining_old_node_fails_debug_3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_joining_old_node_fails.debug.3)
- [test_joining_old_node_fails.dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/16117/testReport/junit/cluster/test_raft_cluster_features/Tests___Unit_Tests___test_joining_old_node_fails_dev_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_joining_old_node_fails.dev.1)
- [test_joining_old_node_fails.dev.2](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/16117/testReport/junit/cluster/test_raft_cluster_features/Tests___Unit_Tests___test_joining_old_node_fails_dev_2) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_joining_old_node_fails.dev.2)
- [test_joining_old_node_fails.dev.3](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/16117/testReport/junit/cluster/test_raft_cluster_features/Tests___Unit_Tests___test_joining_old_node_fails_dev_3) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_joining_old_node_fails.dev.3)
- [test_random_failures[stop_after_bootstrapping_initial_raft_configuration-decommission_node].dev.1](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/16117/testReport/junit/cluster.random_failures/test_random_failures/Tests___Unit_Tests___test_random_failures_stop_after_bootstrapping_initial_raft_configuration_decommission_node__dev_1) [:mag:](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+test_random_failures[stop_after_bootstrapping_initial_raft_configuration-decommission_node].dev.1)
#### Build Details:
- Duration: 8 hr 39 min
- Builder: i-05b6872b810752cc1 (m5d.16xlarge)

bhalevy commented:
In v2 ([a4aa4d7](https://github.com/scylladb/scylladb/commit/a4aa4d74c1d8b22beb8429b4a086f1e41748d516)):
- rebased
- fixed test_raft_cluster_features

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Framework test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/16489/)
:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/16489/)
:white_check_mark: - [Unit Tests Custom](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/16489/)
**The following new/updated tests ran 100 times for each mode:**
:small_blue_diamond: cluster/auth_cluster/test_auth_no_quorum
:small_blue_diamond: cluster/auth_cluster/test_auth_raft_command_split
:small_blue_diamond: cluster/auth_cluster/test_auth_v2_migration
:small_blue_diamond: cluster/auth_cluster/test_raft_service_levels
:small_blue_diamond: cluster/test_aggregation
:small_blue_diamond: cluster/test_automatic_cleanup
:small_blue_diamond: cluster/test_cluster_features
:small_blue_diamond: cluster/test_concurrent_schema
:small_blue_diamond: cluster/test_gossiper
:small_blue_diamond: cluster/test_mutation_schema_change
:small_blue_diamond: cluster/test_mv
:small_blue_diamond: cluster/test_raft_cluster_features
:small_blue_diamond: cluster/test_random_tables
:small_blue_diamond: cluster/test_replace_alive_node
:small_blue_diamond: cluster/test_topology_rejoin
:small_blue_diamond: cluster/test_topology_schema
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/16489/)
#### Build Details:
- Duration: 7 hr 37 min
- Builder: spider1.cloudius-systems.com

bhalevy commented:
@scylladb/scylla-maint please consider merging

pehala commented on test/cluster/auth_cluster/test_auth_no_quorum.py:
```diff
@@ -27,7 +27,7 @@ async def test_auth_no_quorum(manager: ManagerClient) -> None:
         'permissions_update_interval_in_ms': 0,
     }
-    servers = await manager.servers_add(3, config=config)
+    servers = await manager.servers_add(3, config=config, auto_rack_dc="dc1")
```
I do not think this test cares about DCs, so why do we need to define any here? Could it be a default?

I think this would be good followup, no need to change it here

dawmd commented:
I didn't notice there's no backport label here. Can we add `backport/2025.1`? It'll make backporting adjustments in tests (scylladb/scylladb#23490) easier

bhalevy commented:
> I didn't notice there's no backport label here. Can we add `backport/2025.1`? It'll make backporting adjustments in tests (#23490) easier

Can you please prepare a backport branch so we get a feeling if there are any non-trivial merge conflicts?

dawmd commented:
> > I didn't notice there's no backport label here. Can we add `backport/2025.1`? It'll make backporting adjustments in tests (#23490) easier
> 
> Can you please prepare a backport branch so we get a feeling if there are any non-trivial merge conflicts?

Sure, here: https://github.com/scylladb/scylladb/compare/next-2025.1...dawmd:scylladb:pylib-property-2025.1?expand=1. It seems that pretty much all of the conflicts should be trivially resolvable. I can send a patch unless it's better to do it via the bot.
